contents/blog/clickhouse-vs-postgres.md

```diff
@@ -0,0 +1,186 @@
+
+- **Fast Refresh Module.** By blending a static Materialized View and a running log of changes, Postgres’s Fast Refresh Module can simulate an automatically-updated Materialized View. The downsides are that utilizing a log is incompatible with certain fetch requirements (e.g. count), high-write databases could overwhelm the log, and that logs slow down a fetch query.
+ 
+- **Triggers.** You could set a Postgres trigger to refresh a Materialized View on page load. This solution will rarely work unless you have seldom write operations; refreshing the Materialized View completely wipes the original result and will likely bludgeon the CPU. 
+
+![hacking postgres](../images/blog/clickhouse-vs-postgres/hacking-postgres-meme.png) 
+
+In a nutshell, Postgres can be bandaged up to achieve some efficiency that ClickHouse boasts around Materialized Views, but fundamentally, ClickHouse treats Materialized Views as an out-of-the-box benefit with efficiency *and* simplicity as cornerstone value props. 
+
+### Specialized engines vs one-size-fits-all
+ 
+Technically, database engines are nothing new. mySQL has plenty of engines, although it is typically used with just InnoDB. Postgres technically only operates using a single engine, though the Postgres team is building a new engine called [zheap](https://www.percona.com/live/19/sessions/zheap-the-next-generation-storage-engine-for-postgres), specifically designed to optimize the `UPDATE` function. 
+
+The difference between Postgres and ClickHouse engines are similar to the broader comparisons of the databases — Postgres champions a one-size-fits-all approach, while ClickHouse expects users to harness specialized engines when carrying out certain data fetches. 
+
+For ClickHouse, engines are a core feature. ClickHouse should be instructed to utilize a specialized engine depending on your data needs, and that engine could dramatically optimize results. Notably, different materialized views could have different engines, such as `AggregatingMergeTree()` or `SummingMergeTree()`, form-fitted to that materialized view’s purpose.  
+
+ClickHouse also has specialized engines — `MaterializedView`, `Merge`, `Dictionary` etc. — that are used ephemerally to move, merge, or export data. For instance, the MaterializedView engine is able to create a new materialized view much faster than a generalized engine (not to operate it, however; the materialized view will utilize its own engine). 
+
+#### What is Vectorized Query Execution? 
+
+Vectorized Execution, originally championed by the MonetDB team, batches data to achieve bulk processing — a technique previously discussed throughout this article. While there are some third party [Postgres extensions](https://github.com/citusdata/postgres_vectorization_test) to achieve Vectorized Query Execution, ClickHouse is built around it, processing data on every CPU while better utilizing the CPU cache and SIMD CPU instructions.
```

Comment:
@mathexl This section feels a bit random. Would this make more sense as a boxout, or something like that?

Reply:
Probably want to have a section formatted like this:
- Query engine
  - No cost based query planner (deterministic execution)
  - Vectorized execution

Reply:
Good idea.